hs_access_control/models/group.py

```diff
@@ -113,24 +178,97 @@ def members(self):
                                    u2ugp__group=self.group,
                                    u2ugp__privilege__lte=PrivilegeCodes.VIEW)
 
+    @property
+    def viewers(self):
+        """ a viewer is not necessarily a member, due to community influence """
+        return User.objects.filter(
+                Q(is_active=True) &
+                (Q(u2ugp__group__gaccess__active=True,
+                   u2ugp__group=self.group) |
+                 Q(u2ugp__group__gaccess__active=True,
+                   u2ugp__group__g2gcp__community__c2gcp__group__gaccess__active=True,
+                   u2ugp__group__g2gcp__community__c2gcp__group=self.group))).distinct()
+
+    def communities(self):
+        """
+        Return list of communities of which this group is a member.
+
+        :return: list of communities
+
+        """
+        return Community.objects.filter(c2gcp__group=self.group)
```

Comment:
This can also return groups that have been unshared with the community.

Reply:
Nope. Those don't have records in c2gcp.